keras_cv/visualization/plot_image_gallery.py

```diff
@@ -92,27 +91,69 @@ def unpackage_tfds_inputs(inputs):
             "to be true."
         )
 
-    fig = plt.figure(figsize=(cols * scale, rows * scale))
-    fig.tight_layout()  # Or equivalently,  "plt.tight_layout()"
-    plt.subplots_adjust(wspace=0, hspace=0)
-    plt.margins(x=0, y=0)
-    plt.axis("off")
+    if isinstance(images, tf.data.Dataset):
+
+        # Find final dataset batch size
+        sample = next(iter(images.take(1)))
+        if len(sample[image_key].shape) == 3:
+            default_dataset_batch_size = 8
+            images = images.ragged_batch(batch_size=default_dataset_batch_size)
+        elif len(sample[image_key].shape) == 4:
+            default_dataset_batch_size = sample[image_key].shape[0]
+        else:
+            raise ValueError(
+                "plot_image_gallery() expects `tf.data.Dataset` to have TensorShape with length 3 or 4."
+            )
+
+        batches = default_dataset_batch_size
+
+        def unpack_images(inputs):
+            return inputs[image_key]
+
+        images = images.map(unpack_images)
+        images = images.take(batches)
+        images = next(iter(images))
+
+    # Calculate appropriate number of rows and columns
+    if rows is None and cols is None:
```

Comment:
To infer rows/cols based on batch size, we should probably strive for a square instead of a 2xN, which is what we have here (I think?)

What about 
```
rows = math.ceil(batch_size ** 0.5)
cols = math.ceil(batch_size / cols)
```